spec/requests/posts_controller_spec.rb

```diff
@@ -1852,6 +1852,31 @@
             I18n.t("activerecord.errors.models.topic.attributes.base.unable_to_unlist"),
           )
         end
+
+        context "with apply_modifier" do
+          fab!(:group)
+          fab!(:private_group) { Fabricate(:group, users: [admin]) }
+
+          before { Jobs.run_immediately! }
+
+          it "allows mentioning any groups" do
+            private_group.update!(mentionable_level: Group::ALIAS_LEVELS[:members_mods_and_admins])
+            group.add(user)
+            plugin = Plugin::Instance.new
+            modifier = :mentionable_groups
+            proc = Proc.new { Group.all }
+            DiscoursePluginRegistry.register_modifier(plugin, modifier, &proc)
+
+            post "/posts.json",
```

Comment:
Instead of doing a request here you should just do 
```
expect(Group.mentionable).to eq(Group.all)
```

You aren't testing group mentioning here.